cpp/include/cuspatial/types.hpp

```diff
@@ -16,46 +16,37 @@
 
 #pragma once
 
-namespace cuspatial {
+#include <cstdint>
 
-/**
- * @brief A 3D location: latitude, longitude, altitude
- *
- * @tparam T the base type for the coordinates
- */
-template <typename T>
-struct location_3d {
-  T latitude;
-  T longitude;
-  T altitude;
-};
+namespace cuspatial {
 
 /**
- * @brief A 2D Cartesian location (x, y)
+ * @brief A 2D location: latitude and longitude
  *
  * @tparam T the base type for the coordinates
  */
 template <typename T>
-struct coord_2d {
-  T x;
-  T y;
+struct alignas(2 * sizeof(T)) location_2d {
```

Comment:
`location_2d` and `to_location_2d` seems very useful whenever we use SoA for 2d coordinates. In some cases the coordinates are used in planar computes and was not called "latitude" and "longtitude" but "xs" and "ys". Should we make a distinction between geolocation coordinates and cartesian coordinates?

Reply:
yes notice the `coord_2d` that was removed (due to being unused in cuspatial at this time). We can add it back when we start using it.